### PR TITLE
test: integration tests

### PR DESCRIPTION
**Pull request by zac-li**
Added basic test cases.

### CODE REVIEW

**Review by deepankarm — COMMENTED** [review]
Why not combine all routes into one app? It'd deploy all endpoints and they can be tested in one go.

**Comment by zac-li** [comment]
@deepankarm good point, changed. I also added workflow, but can't test it as it's not on `main` yet. I can follow up in a different PR.

File: .github/workflows/integration-tests.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
+name: Integration tests
+
+on:
+  workflow_dispatch:
```
Comment (deepankarm) [inline .github/workflows/integration-tests.yml:4]: This shouldn't be workflow dispatch. It must be pull_request to test on CI

**Review by deepankarm — CHANGES_REQUESTED** [review]

File: .github/workflows/integration-tests.yml (hunk 1 above)
Comment (zac-li) [reply .github/workflows/integration-tests.yml:4]: then it should be part of the `ci.yml`? I can move it over; I was just copying from jcloud's setup, where we need to trigger manually / by schedule, not depending on PR.

Comment (deepankarm) [reply .github/workflows/integration-tests.yml:4]: Yah. This has to be a part of CI

Comment (zac-li) [reply .github/workflows/integration-tests.yml]: Ok, should be fixed by now.

**Review by deepankarm — APPROVED** [review]

**Pull request merged**